code/obj/item/grenades.dm

```diff
@@ -529,7 +529,55 @@ TYPEINFO(/obj/item/old_grenade/singularity)
 
 		qdel(src)
 		return
+/obj/item/old_grenade/foam_dart
+	name = "Foam dart Grenade"
+	desc = "You can make great fights with these and foam dart guns."
+	icon_state = "foam-dart"
+	det_time = 3 SECONDS
+	org_det_time = 3 SECONDS
+	alt_det_time = 6 SECONDS
+	item_state = "fragnade"
+	is_syndicate = 0
+	sound_armed = 'sound/weapons/pindrop.ogg'
+	icon_state_armed = "foam-dart1"
+	var/custom_projectile_type = /datum/projectile/bullet/foamdart/biodegradable
+	var/pellets_to_fire = 25
 
+	prime()
+		var/turf/T = ..()
+		if (T)
+			playsound(T, 'sound/weapons/grenade.ogg', 25, 1)
+			var/obj/overlay/O = new/obj/overlay(get_turf(T))
+			O.anchored = 1
+			O.name = "Explosion"
+			O.layer = NOLIGHT_EFFECTS_LAYER_BASE
+			O.icon = 'icons/effects/64x64.dmi'
+			O.icon_state = "explo_fiery"
```

Comment:
this is really gross and should be a subtype of /obj/overlay instead of doing this

Reply:
how do i do that

Reply:
https://spacestation13.github.io/DMByExample/objs/inheritance.html

Reply:
Not sure why you want a fiery explosion from a foam grenade anyway